group-coordinator/src/test/java/org/apache/kafka/coordinator/group/GroupMetadataManagerTest.java

```diff
@@ -10921,6 +10823,1158 @@ public void testLastConsumerProtocolMemberRebalanceTimeoutInConsumerGroup() {
+            GroupMetadataManagerTestContext.JoinResult joinResult = context.sendClassicGroupJoin(
+                request,
+                true
+            );
+
+            if (replaySuccessfully) {
+                joinResult.appendFuture.complete(null);
+
+                String newMemberId = joinResult.joinFuture.get().memberId();
+                ConsumerGroupMember expectedMember = new ConsumerGroupMember.Builder(newMemberId)
+                    .setMemberEpoch(0)
+                    .setPreviousMemberEpoch(0)
+                    .setInstanceId(instanceId)
+                    .setState(MemberState.UNREVOKED_PARTITIONS)
+                    .setClientId("client")
+                    .setClientHost("localhost/127.0.0.1")
+                    .setSubscribedTopicNames(Collections.singletonList(fooTopicName))
+                    .setRebalanceTimeoutMs(500)
+                    .setSupportedClassicProtocols(request.protocols())
+                    .setPartitionsPendingRevocation(mkAssignment(
+                        mkTopicAssignment(fooTopicId, 0, 1)))
+                    .build();
+
+                List<Record> expectedRecords = Arrays.asList(
+                    // Remove the old static member.
+                    RecordHelpers.newCurrentAssignmentTombstoneRecord(groupId, memberId),
+                    RecordHelpers.newTargetAssignmentTombstoneRecord(groupId, memberId),
+                    RecordHelpers.newMemberSubscriptionTombstoneRecord(groupId, memberId),
+
+                    // Create the new static member.
+                    RecordHelpers.newMemberSubscriptionRecord(groupId, expectedMember),
+                    RecordHelpers.newTargetAssignmentRecord(groupId, newMemberId, Collections.emptyMap()),
+                    RecordHelpers.newTargetAssignmentEpochRecord(groupId, 10),
+                    RecordHelpers.newCurrentAssignmentRecord(groupId, expectedMember)
+                );
+                assertRecordsEquals(expectedRecords, joinResult.records);
+                assertEquals(
+                    new JoinGroupResponseData()
+                        .setMemberId(newMemberId)
+                        .setGenerationId(0)
+                        .setProtocolType(ConsumerProtocol.PROTOCOL_TYPE)
+                        .setProtocolName("range"),
+                    joinResult.joinFuture.get()
+                );
+
+                context.assertSessionTimeout(groupId, newMemberId, request.sessionTimeoutMs());
+                context.assertSyncTimeout(groupId, newMemberId, request.rebalanceTimeoutMs());
+            } else {
+                joinResult.appendFuture.completeExceptionally(new NotLeaderOrFollowerException());
+                assertEquals(
+                    new JoinGroupResponseData()
+                        .setErrorCode(Errors.NOT_COORDINATOR.code()),
+                    joinResult.joinFuture.get()
+                );
+            }
+        }
+    }
```

Comment:
I wonder if we should have a third test case for the static membership: existing member rejoins but with new subscriptions.